docs/source/developer_documentation.rst

```diff
@@ -0,0 +1,126 @@
+-  pre-commit install
+
+Coding Conventions
+------------------
+
+A variety of coding conventions are enforced by automated tools in continuous integeration (black, isort, ruff, numpydoc-validation)
+
+PyRenew Principles
+------------------
+
+-  Variable naming conventions
+
+   -  Use the ``data_`` prefix for (potentially) observed data.
+   -  Use the ``_rv`` suffix for random variables.
+   -  Use the ``observed_`` for the output of sample statements where ``obs`` is a ``data_`` prefixed object.
+   -  Thus, code which may reasonably written like ``infections = infections.sample(x, obs=infections)`` should instead be written ``observed_infections = infections_rv.sample(x, obs=data_infections)``.
+
+-  ``Model`` class conventions
+
+   -  Composing models is discouraged.
+   -  Returning anything from a ``Model`` is discouraged. Instead, sample from models using ``Predictive`` or our ``prior_predictive`` or ``posterior_predictive`` functions.
+   -  Using ``numpyro.deterministic`` within a ``RandomVariable`` is discouraged. Only use at the ``numpyro.deterministic`` ``Model`` level.
+   -  Using default site names in a ``RandomVariable`` is discouraged. Only use default site names at the ``Model`` level.
+   -  Use ``DeterministicVariable``\ s instead of constants within a model.
+
+
+Adding documentation to Sphinx
+------------------------------
+
+The project uses Sphinx to generate documentation. Although code documentation is automatically generated by Sphinx, modules (Python source files) should be explicitly listed. To learn more about Sphinx, visit the `Sphinx documentation <https://www.sphinx-doc.org/en/master/>`__.
+
+
+New module or classes
+~~~~~~~~~~~~~~~~~~~~~
+
+For example, under the ``./docs/source/msei_reference``, the ``index.rst`` file lists the ``HospitalAdmissions`` model by adding the following entry:
+
+.. code-block::
+    Hospital Admissions
+    -------------------
+
+    .. automodule:: pyrenew.latent.hospitaladmissions
+       :members:
+       :undoc-members:
+       :show-inheritance:
+
+This entry tells Sphinx to generate documentation for the ``HospitalAdmissions`` model and its members.
+
+New tutorials
+~~~~~~~~~~~~~
+
+``PyRenew`` tutorials are `quarto documents <https://quarto.org>`__ located under `./model/docs <https://github.com/CDCgov/multisignal-epi-inference/tree/main/model/docs>`__. Tutorials are automatically rendered using GitHub actions (see the Workflow file `here <https://github.com/CDCgov/multisignal-epi-inference/actions/workflows/website.yaml>`__). To make the new tutorial available in the website, developers should follow these steps:
+
+1. Create a new ``quarto`` file in the ``./model/docs`` directory. For instance, the ``example_with_datasets.qmd`` file was added to the repository.
+
+2. Add an entry in the ``./docs/source/tutorials/index.rst``, for example:
+
+.. code-block::
+    .. toctree::
+       :maxdepth: 2
+
+       getting_started
+       example_with_datasets
```

Comment:
I can't make the code blocks work properly. These do not render in the website.

Reply:
You need a blank line after `.. code-block::`